x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/accesscontrol/IndicesAccessControl.java

```diff
@@ -82,20 +84,60 @@ public FieldPermissions getFieldPermissions() {
          *         then this means that there are no document level restrictions
          */
         @Nullable
-        public Set<BytesReference> getQueries() {
-            return queries;
+        public DocumentPermissions getDocumentPermissions() {
+            return documentPermissions;
+        }
+
+        /**
+         * Returns a instance of {@link IndexAccessControl} limited by given index access control
+         * field permissions and document permissions.
```

Comment:
I would prefer that we explained "limited by", e.g.
> i.e. The privileges in the returned <code>IndexAccessControl</code> are the privilege of _this_ object, constrained so that they do not exceed the privileges contained in the provided parameter.